reset.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# The purpose of this script is to make it easy to reset a local onpremise
+# install to a clean state, optionally installing a particular version/ref.
+
+set -euo pipefail
+
+if [ -n "${DEBUG:-}" ]; then
+  set -x
+fi
+
+cd "$(dirname $0)"
+
+
+function confirm () {
+  read -p "$1 [y/n] " confirmation
+  if [ "$confirmation" != "y" ]; then
+    echo "Canceled. 😅"
+    exit
+  fi
+}
+
+# Make sure they mean it.
+confirm "☠️  Warning! 😳 This is highly destructive! 😱 Are you sure you wish to proceed?"
+echo "Okay ... good luck! 😰"
+
+# Hit the reset button.
+docker compose down --volumes
+
+# Remove any remaining (likely external) volumes with name matching '.*sentry.*'.
+for volume in $(docker volume list --format '{{ .Name }}' | grep sentry); do
```

Comment:
Why not
```suggestion
for volume in $(docker volume ls | grep '^sentry\-'; do
```

Should be safer as we know all our volumes are prefixed with `sentry-`. We may even go more conservative and use the list from the volume creation script?

Reply:
I had some volumes locally that were prefixed with `onpremise-sentry-`. Not sure where those came from but that is why I went with the more relaxed match. Since one of the main features of this script is to switch between versions locally and volumes can and do change over time, matching broadly makes more sense to me.

Reply:
>I had some volumes locally that were prefixed with onpremise-sentry-. Not sure where those came from but that is why I went with the more relaxed match.

They come from docker-compose and are removed with the `docker-compose down -v` command.

>Since one of the main features of this script is to switch between versions locally and volumes can and do change over time, matching broadly makes more sense to me.

Even if we add/remove volumes in time the principles for those are:
1. Prefer local volumes so they are managed by `docker-compose`
2. Always prefix global volumes with `sentry-` to avoid issues like this.

I'd even argue we should use local volumes only but it is not easy to migrate.

Reply:
Done in 50c2ab91b466d64e09d6755902d38884df114f0d.